Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -74,9 +74,14 @@ IndexInfo::IndexInfo(std::vector<IndexType> indices, const IndexInfo &parent)
           Kernel::make_unique<Parallel::Communicator>(*parent.m_communicator)) {
   if (const auto parentSpectrumDefinitions = parent.spectrumDefinitions()) {
     m_spectrumDefinitions = Kernel::make_cow<std::vector<SpectrumDefinition>>();
+    const auto &indexSet = parent.makeIndexSet(indices);
     auto &specDefs = m_spectrumDefinitions.access();
-    for (const auto i : parent.makeIndexSet(indices))
-      specDefs.push_back(parentSpectrumDefinitions->operator[](i));
+    specDefs.reserve(specDefs.size() + indexSet.size());
+    std::transform(indexSet.begin(), indexSet.end(),
```

Comment:
Could `cbegin` and `cend` be used here?

Reply:
Yes, if someone were to implement them. `IndexInfo` does not have those methods.

Reply:
Ah, on first glance I assumed `indexSet` was an STL container.